library/std/src/sys/os_str/bytes.rs

```diff
@@ -176,6 +176,11 @@ impl Buf {
         self.inner.extend_from_slice(&s.inner)
     }
 
+    #[inline]
+    pub fn leak<'a>(self) -> &'a mut Slice {
+        unsafe { mem::transmute(self.inner.leak()) }
```

Comment:
These transmutes aren't really correct, but there's a bunch of other transmutes in the same file, so whatever.

Reply:
( They're relying on an assumption of slice representations being equal which is an assumption we've been gradually undermining. )

Reply:
that's fair, after this PR goes live on Nightly, I'll submit another PR replacing those transmutes with abstractions based on `slice::from_raw_parts`

Reply:
Though now that I think about it, I'm not sure how would one construct an unsized type...

Reply:
@its-the-shrimp The key thing is these bits:
https://github.com/rust-lang/rust/blob/fd5777c4c5c8fbc54d33d15afb29479180c532be/library/std/src/sys/os_str/bytes.rs#L155-L169

This is NOT safe!
This is "we get away with it because we're std and if the compiler changes its mind we'll change with it, instantly".